test/integration/requestid_test.go

```diff
@@ -287,3 +292,30 @@ func newAuthorizingServer(t *testing.T, mca *minica.CA) *httptest.Server {
 
 	return srv
 }
+
+func requireCAServerToBeAvailable(t *testing.T, address string, timeout time.Duration) {
+	t.Helper()
+
+	ctx, cancel := context.WithTimeout(context.Background(), timeout)
+	defer cancel()
+
+	for !canConnect(ctx, address) {
+		select {
+		case <-ctx.Done():
+			require.FailNow(t, fmt.Sprintf("CA server failed to start at https://%s within %s", address, timeout.String()))
+		case <-time.After(100 * time.Millisecond):
+		}
```

Comment:
Not sure I understand this `select` statement. Why sleep for 100ms and then return?

Reply:
After 100ms the connection is retried. If it succeeds, the `for` loop is exited, and we continue the test. If it fails to connect at least once before the `ctx` times out, the entire test is failed.